print_all_bug_page.php

```diff
@@ -72,21 +72,11 @@
 
 # Initialize the filter from the cookie, use default if not set
 $t_cookie_value_id = gpc_get_cookie( config_get( 'view_all_cookie' ), '' );
-$t_cookie_value = filter_db_get_filter( $t_cookie_value_id );
-if( is_blank( $t_cookie_value ) ) {
-	$t_filter_cookie_arr = filter_get_default();
-} else {
-	# check to see if new cookie is needed
-	if( !filter_is_cookie_valid() ) {
-		print_header_redirect( 'view_all_set.php?type=0&print=1' );
-	}
-	$t_setting_arr = explode( '#', $t_cookie_value, 2 );
-	$t_filter_cookie_arr = json_decode( $t_setting_arr[1], true );
-}
+$t_filter = filter_get( $t_cookie_value_id, filter_get_default() );
```

Comment:
Should we only create the default filter when needed, rather than constructing it and not using it in most cases?

Reply:
I understand.
note that your comments are on outdated code. Now the offending call is placed in `current_user_get_bug_filter()`
affects `filter_get()` and `filter_temporary_get()`

I think of two options:
1- If the default value is passed as "true", a defaulted filter is built only if needed as return vale inside those functions
2- `filter_get_default()` has a static filter array, so it has to be built only once. Similar to what is done in `filter_get_default_array()`

The default filter concept needs to be reviewed in future works, aiming at: differentiate "empty" from "default" (as system default, or default per user) when implementing filters that only stores the modified values (simplify filter dialogs)

Reply:
@vboctor 
overlooked the simpler solution: it's fixed in e0c048c3bc620adee83400b5f3e70ad71136fb63